src/test/scala/no/gdl/bookapi/service/translation/TranslationServiceTest.scala

```diff
@@ -0,0 +1,157 @@
+/*
+ * Part of GDL book_api.
+ * Copyright (C) 2018 Global Digital Library
+ *
+ * See LICENSE
+ */
+
+package no.gdl.bookapi.service.translation
+
+import io.digitallibrary.language.model.LanguageTag
+import no.gdl.bookapi.integration.crowdin.CrowdinClient
+import no.gdl.bookapi.model.api._
+import no.gdl.bookapi.model.crowdin.CrowdinFile
+import no.gdl.bookapi.model.domain.{InTranslation, InTranslationFile, TranslationStatus}
+import no.gdl.bookapi.{TestData, TestEnvironment, UnitSuite}
+import org.mockito.Mockito._
+import org.mockito.Matchers._
+
+import scala.util.{Failure, Success}
+
+class TranslationServiceTest extends UnitSuite with TestEnvironment {
+
+  val service = new TranslationService
+
+  override def beforeEach = {
+    resetMocks()
+  }
+
+  test("that addTranslation returns a failure if from-language is not supported") {
+    when(readService.withIdAndLanguage(any[Long], any[LanguageTag])).thenReturn(Some(TestData.Api.DefaultBook))
+    when(supportedLanguageService.getSupportedLanguages).thenReturn(Seq(Language("eng", "English")))
+    when(crowdinClientBuilder.forSourceLanguage(any[LanguageTag])).thenReturn(Failure(new RuntimeException("Not supported fromLanguage")))
+
+    val translateResponse = service.addTranslation(TranslateRequest(1, "nob", "eng"))
```

Comment:
This would read better as `TranslateRequest(bookId = 1, fromLanguage = "nob", toLanguage = "eng")` imo.